app/src/main/java/nu/yona/app/ui/YonaActivity.java

```diff
@@ -1233,51 +1234,48 @@ private void checkVPN() {
             public void run() {
                 isToDisplayLogin = false;
                 skipVerification = true;
-                if (YonaApplication.getEventChangeManager().getSharedPreference().getUserPreferences().getString(PreferenceConstant.PROFILE_UUID, "").equals("")) {
+                if (userPreferences.getString(PreferenceConstant.PROFILE_UUID, "").equals("")) {
                     checkFileWritePermission();
-                } else {
-
-                    if(!AppUtils.isVPNConnected(YonaActivity.this))
+                } else if(!AppUtils.isVPNConnected(YonaActivity.this)) {
+                    if (BuildConfig.FLAVOR == "development") {
```

Comment:
Oops, this one I missed. Never compare strings with ``==``. It might work at times, but at other times it'll fail. Change this into:
```
if (BuildConfig.FLAVOR.equals("development")) {
```

Reply:
Done. Corrected.